src/Microsoft.VisualStudio.Services.Agent/ProcessWrapper.cs

```diff
@@ -0,0 +1,295 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent
+{
+    /**/
+    public class ProcessWrapper
+    {
+        public event EventHandler<ProcessDataReceivedEventArgs> OutputDataReceived;
+
+        public event EventHandler<ProcessDataReceivedEventArgs> ErrorDataReceived;
+
+        public event EventHandler Exited;
+
+        private static Encoding OemEncoding = null;
+
+        public TaskCompletionSource<bool> ProcessExitedEvent = new TaskCompletionSource<bool>();
+
+        public IDictionary<String, String> EnvironmentVariables { get; private set; }
+
+        public ProcessWrapper(String workingFolder, String filename, String arguments, IDictionary<String, String> environmentVariables = null)
+        {
+            EnvironmentVariables = environmentVariables;
+            StartInfo = CreateStartInfo(workingFolder, filename, arguments, EnvironmentVariables);
```

Comment:
It would be better to use the existing start info rather than create a new instance. There's subtle differences when creating a new instance that I can't fully remember. I think the event handlers get unregistered, and also IIRC the behavior is also different when the target file doesn't exist (different error message than normal). I don't know what other implications there are, but from the subtle difference I have seen in the past I am led to believe it is just better to use the existing instance.